src/cp/param.h

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2025 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2025 Siddharth Mishra <admin@brightprogrammer.in>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#ifndef CPDEM_param_H
+#define CPDEM_param_H
+
+#include "cp/vec.h"
+#include "demangler_util.h"
+
+/**
+ * Represents a template or function parameter.
+ */
+typedef struct {
+    DemString *name;
+    DemString *suffix;
+    DemString *prefix;
+} Param;
+
+/**
+ * Initialize clone from p_src to p_dst.
+ *
+ * \param p_dst : Pointer to Param object to init clone into.
+ * \param p_src : Pointer to Param object to init clone of.
+ *
+ * \return p_dst on success.
+ * \return NULL otherwise.
+ * */
+#define param_init_clone(p_dst, p_src)                                                             \
+    ((param_init (p_dst) && (p_src)) ? (dem_string_concat ((p_dst)->name, (p_src)->name),          \
+                                        dem_string_concat ((p_dst)->suffix, (p_src)->suffix),      \
+                                        dem_string_concat ((p_dst)->prefix, (p_src)->prefix),      \
+                                        (p_dst)) :                                                 \
+                                       NULL)
```

Comment:
if the purpose is to call this, then maybe just define a function that does this directly in dem_string ?

Reply:
Yep, I'll do that. I used this `init` then `concat` pattern some other places as well. It'll fix those as well...